fuse/post_history.py

```diff
@@ -0,0 +1,310 @@
+    dfa.loc[:, "fc"] = dfa.first_name.fillna("").map(lambda x: x[:1])
+    dfa.loc[:, "lc"] = dfa.last_name.fillna("").map(lambda x: x[:1])
+    return dfa
+
+
+def set_post_uid_2(post):
+    dfa = post[["uid_2", "first_name", "last_name", "middle_name", "agency_2"]]
+    dfa = dfa.rename(columns={"agency_2": "agency"})
+    dfa.loc[:, "fc"] = dfa.first_name.fillna("").map(lambda x: x[:1])
+    dfa.loc[:, "lc"] = dfa.last_name.fillna("").map(lambda x: x[:1])
+    return dfa
+
+
+def set_post_uid_3(post):
+    dfa = post[["uid_3", "first_name", "last_name", "middle_name", "agency_2"]]
+    dfa = dfa.rename(columns={"agency_3": "agency"})
+    dfa.loc[:, "fc"] = dfa.first_name.fillna("").map(lambda x: x[:1])
+    dfa.loc[:, "lc"] = dfa.last_name.fillna("").map(lambda x: x[:1])
+    return dfa
+
+
+def set_post_uid_4(post):
+    dfa = post[["uid_4", "first_name", "last_name", "middle_name", "agency_2"]]
+    dfa = dfa.rename(columns={"agency_4": "agency"})
+    dfa.loc[:, "fc"] = dfa.first_name.fillna("").map(lambda x: x[:1])
+    dfa.loc[:, "lc"] = dfa.last_name.fillna("").map(lambda x: x[:1])
+    return dfa
+
+
+def set_post_uid_5(post):
+    dfa = post[["uid_5", "first_name", "last_name", "middle_name", "agency_2"]]
+    dfa = dfa.rename(columns={"agency_5": "agency"})
+    dfa.loc[:, "fc"] = dfa.first_name.fillna("").map(lambda x: x[:1])
+    dfa.loc[:, "lc"] = dfa.last_name.fillna("").map(lambda x: x[:1])
+    return dfa
+
+
+def set_post_uid_6(post):
+    dfa = post[["uid_6", "first_name", "last_name", "middle_name", "agency_2"]]
+    dfa = dfa.rename(columns={"agency_6": "agency"})
+    dfa.loc[:, "fc"] = dfa.first_name.fillna("").map(lambda x: x[:1])
+    dfa.loc[:, "lc"] = dfa.last_name.fillna("").map(lambda x: x[:1])
+    return dfa
+
+
+def set_post_uid_7(post):
+    dfa = post[["uid_7", "first_name", "last_name", "middle_name", "agency_2"]]
+    dfa = dfa.rename(columns={"agency_7": "agency"})
+    dfa.loc[:, "fc"] = dfa.first_name.fillna("").map(lambda x: x[:1])
+    dfa.loc[:, "lc"] = dfa.last_name.fillna("").map(lambda x: x[:1])
+    return dfa
+
+
+def set_post_uid_8(post):
+    dfa = post[["uid_8", "first_name", "last_name", "middle_name", "agency_2"]]
+    dfa = dfa.rename(columns={"agency_8": "agency"})
+    dfa.loc[:, "fc"] = dfa.first_name.fillna("").map(lambda x: x[:1])
+    dfa.loc[:, "lc"] = dfa.last_name.fillna("").map(lambda x: x[:1])
+    return dfa
+
+
+def set_personnel_uid(personnel):
+    dfb = personnel[["uid", "first_name", "last_name", "middle_name", "agency"]]
+    dfb.loc[:, "fc"] = dfb.first_name.fillna("").map(lambda x: x[:1])
+    dfb.loc[:, "lc"] = dfb.last_name.fillna("").map(lambda x: x[:1])
+    return dfb
+
+
+def match_uid(dfb, post, dfa=None):
+    matcher = ThresholdMatcher(
+        ColumnsIndex(["fc"]),
+        {
+            "first_name": JaroWinklerSimilarity(),
+            "last_name": JaroWinklerSimilarity(),
+            "agency": JaroWinklerSimilarity(),
```

Comment:
If the `agency` column can be exactly matched (they should be), then just put it in the `ColumnsIndex`.